components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -1293,7 +1272,10 @@ AtmosphereOutput::create_diagnostic (const std::string& diag_field_name) {
       } else if (units=="mb" or units=="Pa" or units=="hPa") {
         diag_name = "FieldAtPressureLevel";
         diag_avg_cnt_name = "_" + tokens[1]; // Set avg_cnt tracking for this specific slice
-        m_track_avg_cnt = true; // If we have pressure slices we need to be tracking the average count.
+
+        // If we have pressure slices we need to be tracking the average count,
```

Comment:
We don't seem to mask for FieldAtHeight above sealevel, but presumably, 2m above sealevel should be masked in mountainous areas... right?  Maybe something to consider for a different PR.

Reply:
Ah, yes, `FieldAtHeight` with "sealevel" as reference would trigger filling, I think. But I don't think we have that PR merged yet. Perhaps you could do the same for `FieldAtHeight` in your PR, if the ref is "sealevel" and the height is less than 9km?

Reply:
I think the current FieldAtHeight diagnostic extrapolates, which is why we don't actually have an issue.  We would need to change the vertical interpolation routine in the `run_impl` to mask output.  For the sake of getting the FieldAtHeight PR in before the decadal runs I think it would be better to make this change in a separate PR.